subprojects/docs/src/samples/java-feature-variant/incompatible-variants/groovy/producer/build.gradle

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id 'java-library'
+}
+
+repositories {
+    jcenter()
+}
+
+// tag::producer[]
+java {
+    registerFeature('mysqlSupport') {
+        usingSourceSet(sourceSets.main)
```

Comment:
Should we not _default_ to main?

Reply:
I'm not sure we really want to encourage mixing sources. The only reason it's used here is because it's commonly done this way in the Java world.